src/react/admin.js

```diff
@@ -1,5 +1,63 @@
 import React from 'react'
-import ReactDOM from 'react-dom'
+import { render } from 'react-dom'
+import axios from 'axios'
+import get from 'lodash.get'
+
+// i18n (Note: add new languages here and in addLocaleData)
+import { IntlProvider, addLocaleData } from 'react-intl'
+import en from 'react-intl/locale-data/en'
+
+// Components
 import App from './components/app'
 
-ReactDOM.render(<App wpGlobals={wpGlobals} />, document.getElementById('plugin-name-app'))
+// logging and debugging
+const debug = require('debug')('Admin')
+
+// Add Local data (for each language) and define user's language.
+// inspired by
+// - https://medium.freecodecamp.org/internationalization-in-react-7264738274a0
+addLocaleData([...en])
+const language = get(navigator, 'languages[0]') || navigator.language || navigator.userLanguage
+const languageWithoutRegionCode = language.toLowerCase().split(/[_-]+/)[0]
+debug('language', language) /* eslint-disable-line */
+debug('languageWithoutRegionCode', languageWithoutRegionCode) /* eslint-disable-line */
+
+// TODO
+// - IntlProvider w/ Redux:
+//   - https://stackoverflow.com/questions/35776663/react-intl-multi-language-app-changing-languages-and-translations-storage/40603629
+//   - https://github.com/yahoo/react-intl/issues/243#issuecomment-165924794
+// - try: https://github.com/ratson/react-intl-redux
+const renderApp = async () => {
+  try {
+    const result = await axios.get(
+      `${wpGlobals.asset_path}/languages/${languageWithoutRegionCode}.json`,
```

Comment:
does this handle the case, if the user has set a language which is not supported/available as translation?
would this respond with a 404 and then be handled in `catch`? if so, we could prevent this by only making the request if a language is already added via `addLocaleData`?